luaui/Widgets/api_analytics.lua

```diff
@@ -0,0 +1,96 @@
+local widget = widget --- @type Widget
+
+function widget:GetInfo()
+	return {
+		name    = "Analytics API",
+		desc    = "Provides an API for sending analytics events using SendLuaUIMsg ",
+		author  = "uBdead",
+		date    = "Oct 2025",
+		license = "GPL-v2",
+		layer   = -1,
+		enabled = true,
+	}
+end
+
+local spGetPlayerInfo = Spring.GetPlayerInfo
+local spGetMyPlayerID = Spring.GetMyPlayerID
+local spGetGameFrame = Spring.GetGameFrame
+local spLog = Spring.Log
+local spSendLuaUIMsg = Spring.SendLuaUIMsg
+
+local pendingAnalytics = {}
+
+local function analyticsCoroutine(eventType, eventData)
+	local myPlayerName = spGetPlayerInfo(spGetMyPlayerID())
+	local frameNumber = spGetGameFrame()
+	local jsondict = {}
+	-- Add static fields
+	jsondict.eventtype = eventType
+	jsondict.username = myPlayerName
+	jsondict.frame = frameNumber
+
+	-- Add eventData incrementally
+	if type(eventData) == "table" then
+		local keys = {}
+		for k in pairs(eventData) do keys[#keys+1] = k end
+		local i = 1
+		while i <= #keys do
+			local k = keys[i]
+			if jsondict[k] == nil then
+				jsondict[k] = eventData[k]
+			else
+				spLog("Analytics API: Key conflict for '" .. k .. "'. Skipping this key.", Spring.LOG_WARNING)
```

Comment:
1) spLog's first arg is some sort of "category", the actual log is the third and onwards
2) `Spring.LOG_WARNING` -> `LOG.WARNING`
3) avoid the `..` operator inside spLog and spEcho where possible, this is because `..` will crash on types that aren't concatenable (I haven't looked so it may be safe in this case, but it's a good convention anyway because people copypaste code). both these functions are variadic and can print the arg on its own though, e.g.
```diff
  local bar, baz = nil, {}
- Echo("foo " .. bar .. ", " .. baz) -- crashes, can't ..table or ..nil
+ Echo("foo", bar, baz) -- prints "foo, nil, <table>"
```

So ultimately something like:
```suggestion
				spLog("Analytics API", LOG.WARNING, "Key conflict, skipping", k)
```